src/test/java/gov/usgs/wma/waterdata/LoadTimeSeriesTest.java

```diff
@@ -35,35 +36,74 @@ public void setupLoadTimeSeries() {
 		genericTimeSeries2 = new TimeSeries();
 	}
 
+	// tests for fails
+	@Test
+	public void testNullId() {
+		request.setUniqueId(null);
+		ResultObject result = loadTimeSeries.processRequest(request);
+
+		assertNotNull(result);
+		assertEquals(LoadTimeSeries.STATUS_FAIL, result.getStatus());
+		assertEquals(LoadTimeSeries.FAIL_MESSAGE_NULL_UNIQUE_ID, result.getFailMessage());
+		assertNull(result.getCount());
+		assertThrows(RuntimeException.class, () -> {
+			loadTimeSeries.apply(request);
+		}, "should have thrown an exception but did not");
+	}
+
 	@Test
-	public void testNotFound() {
+	public void testNoRecordsFound() {
 		// no time series data found
 		when(transformDao.getTimeSeries(anyString())).thenReturn(genericTimeSeriesList);
-		// nothing is deleted
-		when(observationDao.deleteTimeSeries(anyString())).thenReturn(0);
-		// nothing is inserted
+		ResultObject result = loadTimeSeries.processRequest(request);
+
+		assertNotNull(result);
+		assertEquals(LoadTimeSeries.STATUS_FAIL, result.getStatus());
+		assertEquals(LoadTimeSeries.FAIL_MESSAGE_NO_RECORDS, result.getFailMessage());
+		assertNull(result.getCount());
+		assertThrows(RuntimeException.class, () -> {
+			loadTimeSeries.apply(request);
+		}, "should have thrown an exception but did not");
+	}
+
+	@Test
+	public void testFoundGenericFailedInsert() {
+		genericTimeSeriesList.add(genericTimeSeries1);
+		genericTimeSeriesList.add(genericTimeSeries2);
+		when(transformDao.getTimeSeries(anyString())).thenReturn(genericTimeSeriesList);
+		// delete succeeds
+		when(observationDao.deleteTimeSeries(anyString())).thenReturn(2);
+		// insert fails
 		when(observationDao.insertTimeSeries(any())).thenReturn(0);
-		ResultObject result = loadTimeSeries.apply(request);
+		ResultObject result = loadTimeSeries.processRequest(request);
+
 		assertNotNull(result);
-		Integer expectedRowsInsertedCount = null;
-		assertEquals(expectedRowsInsertedCount, result.getCount());
 		assertEquals(LoadTimeSeries.STATUS_FAIL, result.getStatus());
+		assertEquals(LoadTimeSeries.FAIL_MESSAGE_INSERT_FAILED, result.getFailMessage());
+		assertEquals(0, result.getCount());
+		// throws exception
+		assertThrows(RuntimeException.class, () -> {
+			loadTimeSeries.apply(request);
+		}, "should have thrown an exception but did not");
 	}
 
+	// tests for successes
 	@Test
 	public void testFoundGeneric() {
 		genericTimeSeriesList.add(genericTimeSeries1);
 		genericTimeSeriesList.add(genericTimeSeries2);
-		// happy path - 2 time series returned
+		// 2 time series returned
```

Comment:
Technically, aren't these time steps within a single time series. It is a nit, but we need to be as clear as possible so the next person doesn't get confused.